htslib/vcf.h

```diff
@@ -284,11 +284,11 @@ typedef struct {
     typedef htsFile vcfFile;
     #define bcf_open(fn, mode) hts_open((fn), (mode))
     #define vcf_open(fn, mode) hts_open((fn), (mode))
-    #define bcf_close(fp) hts_close(fp)
-    #define vcf_close(fp) hts_close(fp)
+    #define bcf_close(fp) hts_close(fp) HTS_RESULT_USED
+    #define vcf_close(fp) hts_close(fp) HTS_RESULT_USED
```

Comment:
My bad, sorry. Wanted to add directly `hts_close()` but did not have time to fix all of samtools unchecked warnings.

Reply:
Yep, maybe one of the test programs should use the macro to catch anything like this:
```diff
+++ b/test/test-vcf-api.c
@@ -45 +45 @@ void write_bcf(char *fname)
-    htsFile *fp    = hts_open(fname,"wb");
+    htsFile *fp    = bcf_open(fname,"wb");
@@ -169 +169 @@ void write_bcf(char *fname)
-    if ( (ret=hts_close(fp)) )
+    if ( (ret=bcf_close(fp)) )
```

I can't find any emails from the time, but I seem to recall that we didn't add this to `hts_close()` when we did the rest, on the basis that if you check all your `sam_read1()` calls there's nothing to be gained from checking `hts_close()`. Or something like that…

Reply:
Oops, should have caught the `#defines`.

`hts_close()` and close functions in general were not done because there's plenty of cases where checking the return value is pointless - notably when cleaning up after other errors have been detected.  It does need to be checked on the main path though, as errors may not be reported until the final flush of the data.  You just have to remember to do it without prompting from the compiler.